File: app/src/main/java/ru/komcity/mobile/forum/SubForumAdapter.java
package ru.komcity.mobile.forum;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import java.util.List;
import butterknife.BindView;
import butterknife.ButterKnife;
import ru.komcity.mobile.R;

public class SubForumAdapter extends RecyclerView.Adapter<SubForumAdapter.SubForumViewHolder> {
    private List<Object> feedItemList;
    private Context mContext;
    private ForumClickListener onItemClickListener;

    public SubForumAdapter(Context context, List<Object> feedItemList) {
        this.feedItemList = feedItemList;
        this.mContext = context;
    }

    @Override
    public SubForumViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.subforum_item, null);
        SubForumViewHolder viewHolder = new SubForumViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(SubForumViewHolder holder, int position) {
        final ForumItem item = (ForumItem)feedItemList.get(position);

        if (item != null) {
            //Setting text view title
            holder.lbl_title_forum.setText(item.getTitle());
            holder.lbl_date_forum.setText(item.getDescription());
            holder.lbl_count_forum.setText(item.getCountReplic().trim() + item.getCountTheme().trim());

            View.OnClickListener listener = new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    onItemClickListener.onItemClick(item);
                }
            };
            holder.lbl_title_forum.setOnClickListener(listener);
            holder.lbl_date_forum.setOnClickListener(listener);
            holder.lbl_count_forum.setOnClickListener(listener);
        }
    }

    @Override
    public int getItemCount() {
        return (null != feedItemList ? feedItemList.size() : 0);
    }


    class SubForumViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.lbl_title)  TextView lbl_title_forum;
        @BindView(R.id.lbl_lastdate) TextView lbl_date_forum;
        @BindView(R.id.lbl_countmessage)  TextView lbl_count_forum;

        public SubForumViewHolder(View view) {
            super(view);
            ButterKnife.bind(this, view);
        }
    }

    public ForumClickListener getOnItemClickListener() {
        return onItemClickListener;
    }

    public void setOnItemClickListener(ForumClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
    }
}

File: app/src/main/java/ru/komcity/mobile/ui/adapter/SubForumAdapter.kt
package ru.komcity.mobile.ui.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import ru.komcity.mobile.databinding.ItemSubForumBinding
import ru.komcity.mobile.viewModel.SubForumItem

/**
 * Created by Aleksey on 2020.03.01
 * <p>
 * Адаптер для списка подфорумов
 */
class SubForumAdapter(private val items: List<SubForumItem>,
                      private val listener: (title: String, forumId: String) -> Unit) : RecyclerView.Adapter<SubForumAdapter.ViewHolder>() {

    private var _binding: ItemSubForumBinding? = null
    private val binding get() = _binding!!

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        _binding = ItemSubForumBinding.inflate(LayoutInflater.from(parent.context), parent, true)
        return ViewHolder(binding.root)
    }

    override fun getItemCount(): Int {
        return items.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(items[position])
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind(item: SubForumItem) {
            with(binding) {
                tvTitle.text = item.forumName
                tvThemeCount.text = item.countTheme
                tvReplicCount.text = item.countReplic
                tvDescription.text = item.description
                initClickListener(root, item)
            }
        }

        private fun initClickListener(itemView: View, item: SubForumItem) {
            with(binding) {
                tvTitle.setOnClickListener {listener(item.forumName, item.linkForum)  }
                tvThemeCount.setOnClickListener {listener(item.forumName, item.linkForum)  }
                tvReplicCount.setOnClickListener {listener(item.forumName, item.linkForum)  }
                tvDescription.setOnClickListener {listener(item.forumName, item.linkForum)  }
                root.setOnClickListener { listener(item.forumName, item.linkForum) }
            }
        }
    }
}